presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestPrestoDatabaseMetaData.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import com.facebook.presto.execution.QueryInfo;
+import com.facebook.presto.server.testing.TestingPrestoServer;
+import com.facebook.presto.spi.QueryId;
+import io.airlift.log.Logging;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableSet;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestPrestoDatabaseMetaData
+{
+    private TestingPrestoServer server;
+
+    private Connection connection;
+
+    @BeforeClass
+    public void setupServer()
+            throws Exception
+    {
+        Logging.initialize();
+        server = new TestingPrestoServer();
+    }
+
+    @AfterClass
+    public void tearDownServer()
+    {
+        server.close();
+    }
+
+    @SuppressWarnings("JDBCResourceOpenedButNotSafelyClosed")
+    @BeforeMethod
+    public void setup()
+            throws Exception
+    {
+        connection = createConnection();
+    }
+
+    @AfterMethod
+    public void tearDown()
+            throws SQLException
+    {
+        connection.close();
+    }
+
+    @Test
+    public void testPassEscapeInMetaDataQuery()
+            throws Exception
+    {
+        DatabaseMetaData metaData = connection.getMetaData();
+
+        Set<String> queries = captureQueries(() -> {
+            try (ResultSet resultSet = metaData.getColumns("blackhole", "defau" + metaData.getSearchStringEscape() + "_t", null, null)) {
+                assertFalse(resultSet.next(), "There should be no results");
+            }
+            return null;
+        });
+
+        assertEquals(queries.size(), 1, "Expected exactly one query, got " + queries.size());
+        String query = getOnlyElement(queries);
+
+        assertTrue(
```

Comment:
Use `assertContains` from Airlift `Assertions` class, then you don't need the custom message.